lib/streamlit/web/cli.py

```diff
@@ -253,15 +254,16 @@ def cache():
 
 @cache.command("clear")
 def cache_clear():
-    """Clear st.cache, st.memo, and st.singleton caches."""
+    """Clear st.cache, st.cache_data, and st.cache_resource caches."""
     result = legacy_caching.clear_cache()
     cache_path = legacy_caching.get_cache_path()
     if result:
         print(f"Cleared directory {cache_path}.")
     else:
         print(f"Nothing to clear at {cache_path}.")
 
-    caching.cache_data.clear()
+    cache_storage_manager = get_cache_storage_manager()
+    cache_storage_manager.clear_all()
     caching.cache_resource.clear()
```

Comment:
Actually, `streamlit cache clear` work from another python process doesn't have access to in_memory cache (that was true for previous implementation, and remains true for current. 

It removes only persist cache (in our case only files from disk)

Related to this maybe we need to remove `caching.cache_resource.clear()` call from here at all, and also rework documentation to not mislead users.

Reply:
Yes, we should not change the behavior or add anything new or complex here. If it seems like any regression lets discuss (but sounds like it's not any).

Agree if we can easily update the docs / code to be more clear about the behavior, that's a nice to have.